lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -199,19 +199,42 @@ public function tablesExist($tableNames)
         return count($tableNames) === count(array_intersect($tableNames, array_map('strtolower', $this->listTableNames())));
     }
 
+    /**
+    * Returns true if all the given tables exist. This method ignores filters.
+    *
+    * @param string[] $tableNames
+    *
+    * @return bool
+    */
+    public function allTablesExistInSchema(array $tableNames) : bool
+    {
+       $tableNames = array_map('strtolower', $tableNames);
+
+       return count($tableNames) === count(array_intersect($tableNames, array_map('strtolower', $this->listAllTableNames())));
```

Comment:
Why are we using `strtolower here`?

Reply:
Primarily because it was an earlier code, so it was in the tablesExist method. The table name can be both uppercase and lowercase.

But I do not see what the consequences would be if we changed this now.

Reply:
Oh right. Sounds like you could refactor this in a private method, right? `isTableListSubsetOf($leftSet, $rightSet): bool`, something like that…

Reply:
You mean that?

```php
     /**
     * Returns true if all the given tables exist.
     *
     * @param string|string[] $tableNames
     *
     * @return bool
     */
    public function tablesExist($tableNames)
    {
        return $this->isTableListSubsetOf(
          array_map('strtolower', (array) $tableNames),
          array_map('strtolower', $this->listTableNames())
        );
    }

    /**
    * Returns true if all the given tables exist. This method ignores filters.
    *
    * @param string[] $tableNames
    *
    * @return bool
    */
    public function allTablesExistInSchema(array $tableNames) : bool
    {
        return $this->isTableListSubsetOf(
          array_map('strtolower', $tableNames),
          array_map('strtolower', $this->listAllTableNames())
        );
    }

    private function isTableListSubsetOf(array $leftSet, array $rightSet) : bool
    {
        return count($leftSet) === count(array_intersect($leftSet, $rightSet));
    }
```

Reply:
I mean having the `array_map` calls in there too…